### ISSUE TITLE
Consistent crash ("heap-use-after-free") for any video with latest Git HEAD

### ISSUE THREAD

**Issue by forthrin** [mpv-player/mpv#13823]
### Important Information

mpv version: git-2024-04-06-f4c839922c
FFmpeg version: git-2024-04-05-6f13f5dd
System Version: macOS 14.3.1 (23D60)
Model Identifier: MacBookAir10,1

### Reproduction steps

1. `$ mpv --no-config any-video-it-seems.mp4`
2. Quit

### Log file

```
==51588==ERROR: AddressSanitizer: heap-use-after-free on address 0x00010b72ea10 at pc 0x00010227325c bp 0x00016decdc70 sp 0x00016decdc68
READ of size 8 at 0x00010b72ea10 thread T3
    #0 0x102273258 in mp_msg_level msg.c:184
    #1 0x102278f18 in mp_msg_test msg.h:60
    #2 0x102276bec in mp_msg_va msg.c:514
    #3 0x1020f8988 in LogHelper.send(message:type:) log_helper.swift:59
    #4 0x1020f81d4 in LogHelper.verbose(_:) log_helper.swift:37
    #5 0x1020eae60 in AppHub.initInput(_:) app_hub.swift:73
    #6 0x1020eaecc in @objc AppHub.initInput(_:) <compiler-generated>
    #7 0x102df3b28 in cocoa_set_input_context app_bridge.m:105
    #8 0x102899250 in mp_destroy main.c:184
    #9 0x1028a14c4 in mpv_main main.c:465
    #10 0x1020e8b80 in closure #1 in variable initialization expression of Application.playbackThread application.swift:98
    #11 0x1020e8b98 in @objc closure #1 in variable initialization expression of Application.playbackThread <compiler-generated>
    #12 0x1873c2030 in _pthread_start+0x84 (libsystem_pthread.dylib:arm64e+0x7030)
    #13 0x1873bce38 in thread_start+0x4 (libsystem_pthread.dylib:arm64e+0x1e38)

0x00010b72ea10 is located 80 bytes inside of 272-byte region [0x00010b72e9c0,0x00010b72ead0)
freed by thread T0 here:
    #0 0x1084eb380 in wrap_free+0x98 (libclang_rt.asan_osx_dynamic.dylib:arm64e+0x53380)
    #1 0x102deb54c in ta_free ta.c:246
    #2 0x102dec1b0 in ta_free_children ta.c:231
    #3 0x102deb530 in ta_free ta.c:243
    #4 0x102747c7c in mp_destroy_client client.c:487
    #5 0x1027450c0 in mpv_destroy client.c:543
    #6 0x1020ee274 in EventHelper.handle(event:) event_helper.swift:117
    #7 0x1020ede7c in EventHelper.eventLoop() event_helper.swift:102
    #8 0x1020edd30 in closure #1 in closure #1 in variable initialization expression of EventHelper.wakeup event_helper.swift:96
    #9 0x1020e7dd0 in thunk for @escaping @callee_guaranteed () -> () <compiler-generated>
    #10 0x1084ea350 in __wrap_dispatch_async_block_invoke+0xc0 (libclang_rt.asan_osx_dynamic.dylib:arm64e+0x52350)
    #11 0x18720ecb4 in _dispatch_call_block_and_release+0x1c (libdispatch.dylib:arm64e+0x1cb4)
    #12 0x18721090c in _dispatch_client_callout+0x10 (libdispatch.dylib:arm64e+0x390c)
    #13 0x18721efa4 in _dispatch_main_queue_drain+0x3d4 (libdispatch.dylib:arm64e+0x11fa4)
    #14 0x18721ebbc in _dispatch_main_queue_callback_4CF+0x28 (libdispatch.dylib:arm64e+0x11bbc)
    #15 0x1874e0ea0 in __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__+0xc (CoreFoundation:arm64e+0xbeea0)
    #16 0x18749e75c in __CFRunLoopRun+0x7c8 (CoreFoundation:arm64e+0x7c75c)
    #17 0x18749d938 in CFRunLoopRunSpecific+0x25c (CoreFoundation:arm64e+0x7b938)
    #18 0x191a66444 in RunCurrentEventLoopInMode+0x120 (HIToolbox:arm64e+0x30444)
    #19 0x191a66280 in ReceiveNextEventCommon+0x284 (HIToolbox:arm64e+0x30280)
    #20 0x191a65fd8 in _BlockUntilNextEventMatchingListInModeWithFilter+0x48 (HIToolbox:arm64e+0x2ffd8)
    #21 0x18ac7cecc in _DPSNextEvent+0x290 (AppKit:arm64e+0x39ecc)
    #22 0x18b467ee8 in -[NSApplication(NSEventRouting) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]+0x2c8 (AppKit:arm64e+0x824ee8)
    #23 0x18ac70378 in -[NSApplication run]+0x1d8 (AppKit:arm64e+0x2d378)
    #24 0x1020e8fc0 in Application.main(_:_:) application.swift:113
    #25 0x1020e91d4 in @objc Application.main(_:_:) <compiler-generated>
    #26 0x102df3bc8 in cocoa_main app_bridge.m:120
    #27 0x102ed2dec in main main-fn-mac.c:6
    #28 0x1870410dc  (<unknown module>)

previously allocated by thread T3 here:
    #0 0x1084eb610 in wrap_calloc+0x9c (libclang_rt.asan_osx_dynamic.dylib:arm64e+0x53610)
    #1 0x102dea31c in ta_zalloc_size ta.c:156
    #2 0x10227b674 in mp_log_new msg.c:590
    #3 0x1020eaab4 in AppHub.initMpv(_:) app_hub.swift:58
    #4 0x1020eadec in @objc AppHub.initMpv(_:) <compiler-generated>
    #5 0x102df3b5c in cocoa_set_mpv_handle app_bridge.m:110
    #6 0x10289ecb8 in mp_initialize main.c:394
    #7 0x1028a097c in mpv_main main.c:430
    #8 0x1020e8b80 in closure #1 in variable initialization expression of Application.playbackThread application.swift:98
    #9 0x1020e8b98 in @objc closure #1 in variable initialization expression of Application.playbackThread <compiler-generated>
    #10 0x1873c2030 in _pthread_start+0x84 (libsystem_pthread.dylib:arm64e+0x7030)
    #11 0x1873bce38 in thread_start+0x4 (libsystem_pthread.dylib:arm64e+0x1e38)

Thread T3 created by T0 here:
    #0 0x1084e41b0 in wrap_pthread_create+0x54 (libclang_rt.asan_osx_dynamic.dylib:arm64e+0x4c1b0)
    #1 0x1020e8ec4 in Application.main(_:_:) application.swift:111
    #2 0x1020e91d4 in @objc Application.main(_:_:) <compiler-generated>
    #3 0x102df3bc8 in cocoa_main app_bridge.m:120
    #4 0x102ed2dec in main main-fn-mac.c:6
    #5 0x1870410dc  (<unknown module>)

SUMMARY: AddressSanitizer: heap-use-after-free msg.c:184 in mp_msg_level
Shadow bytes around the buggy address:
  0x00010b72e780: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x00010b72e800: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
  0x00010b72e880: fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd
  0x00010b72e900: fd fd fd fd fd fd fd fd fd fa fa fa fa fa fa fa
  0x00010b72e980: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
=>0x00010b72ea00: fd fd[fd]fd fd fd fd fd fd fd fd fd fd fd fd fd
  0x00010b72ea80: fd fd fd fd fd fd fd fd fd fd fa fa fa fa fa fa
  0x00010b72eb00: fa fa fa fa fa fa fa fa fd fd fd fd fd fd fd fd
  0x00010b72eb80: fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd fd
  0x00010b72ec00: fd fd fd fd fd fd fd fd fd fd fa fa fa fa fa fa
  0x00010b72ec80: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==51588==ABORTING
```

**Comment by Akemi** [comment]
will be fixed with #13843. also on a side note, your referenced mpv commit f4c839922c isn't on master and i can't find it at all.

**Comment by forthrin** [comment]
Immediate tests indicate success. Also thanks for notifying about Git anomaly.
